### PR TITLE
feat(dataset): design structure and interfaces of Dataset

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataset/version.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            status: The draft status which includes "OPEN", "CLOSED", "COMMITTED", "ALL" and None.
+                    where None means listing open drafts.
+            branch_name: The branch name.
+
+        Return:
+            The PagingList of :class:`drafts<.Draft>`.
+
+        """
+
+    def update_draft(
```
Comment (linjiX) [inline graviti/dataset/version.py:85]: will it be better we move the draft related operation into `Draft` instance?
```python
draft = dataset.get_draft()
draft.update()
draft.close()
...
```

Also consider commit branch and tag?

File: graviti/dataset/version.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+class VersionControlMixin:  # pylint: disable=too-many-public-methods
+    """A mixin class supporting version control methods."""
+
+    _url: str
+    _access_key: str
+    _dataset_id: str
+    _status: Status
+
+    def checkout(self, revision: Optional[str] = None, draft_number: Optional[int] = None) -> None:
```
Comment (linjiX) [inline graviti/dataset/version.py:23]: will it be better to separate `checkout` and `checkout_draft` as two methods?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataset/graviti.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    Arguments:
+        access_key: User's access key.
+        url: The host URL of the gas website.
+
+    """
+
+    def __init__(
+        self,
+        access_key: str,
+        url: str = "",
```
Comment (Lee-000) [inline graviti/dataset/graviti.py:27]: Should `Graviti` have `access_key` and `url` properties?

**Review by AChenQ — APPROVED** [review]

File: graviti/dataset/dataset.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        """Return the config name of the dataset.
+
+        Return:
+            The config name of the dataset.
+
+        """
+
+    @property
+    def branches(self) -> BranchManager:
+        """Get class :class:`~graviti.manager.dataset.BranchManager`.
```
Comment (Lee-000) [inline graviti/dataset/dataset.py:93]: Get `class` or the `instance`?

File: graviti/dataset/dataset.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    def checkout_draft(self, draft_number: int) -> None:
+        """Checkout to a draft.
+
+        Arguments:
+            draft_number: The draft number.
+
+        """
+
+    def commit(self, title: str, description: str = "", *, tag: Optional[str] = None) -> None:
+        """Commit the draft.
```
Comment (Lee-000) [inline graviti/dataset/dataset.py:145]: `Commit the current draft` for better understanding?

File: graviti/dataset/manager.py [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+from tensorbay.client.lazy import PagingList
+from tensorbay.client.struct import Branch, Commit, Tag
+
+from graviti.dataset.dataset import Dataset
+from graviti.dataset.struct import Draft
+
+
+class DatasetManager:
+    """This class defines the operations on the dataset in Graviti.
```
Comment (Lee-000) [inline graviti/dataset/manager.py:18]: `in Graviti` or `on Graviti`?

File: graviti/dataset/manager.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+        pass
+
+    def create(self, name: str, revision: Optional[str] = None) -> Branch:
+        """Create a branch.
+
+        Arguments:
+            name: The branch name.
+            revision: The information to locate the specific commit, which can be the commit id,
+                the branch name, or the tag name.
+                If the revision is not given, create the branch based on the current commit.
```
Comment (Lee-000) [inline graviti/dataset/manager.py:97]: How to get the current commit?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

**Review by Lee-000 — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**